operator/src/main/java/oracle/kubernetes/operator/helpers/JobHelper.java

```diff
@@ -523,12 +525,23 @@ public NextAction onSuccess(Packet packet, CallResponse<String> callResponse) {
 
         final V1Job domainIntrospectorJob = packet.getValue(DOMAIN_INTROSPECTOR_JOB);
         if (severeStatuses.isEmpty()) {
+          if (!isDomainIntrospectionComplete(callResponse)) {
+            LOGGER.severe(DOMAIN_INTROSPECTION_INCOMPLETE, callResponse.getResult());
+            severeStatuses.add(LOGGER.formatMessage(DOMAIN_INTROSPECTION_INCOMPLETE, callResponse.getResult()));
+            return handleFailure(packet, domainIntrospectorJob);
+          }
           return doNext(createRemoveSelectedFailuresStep(getNext(), INTROSPECTION), packet);
         } else {
           return handleFailure(packet, domainIntrospectorJob);
         }
       }
 
+      @NotNull
+      private Boolean isDomainIntrospectionComplete(CallResponse<String> callResponse) {
+        return Optional.ofNullable(callResponse).map(CallResponse::getResult)
+            .map(r -> r.contains(DOMAIN_INTROSPECTION_COMPLETE)).orElse(false);
```

Comment:
The change looks good to me, except please also consider:
- Minimally adding a comment near the key line in the introspector script, something like: "Important: The following trace's wording must not be changed without a corresponding change to DOMAIN_INTROSPECTION_COMPLETE in the operator."
- Perhaps also adding a unit test that verifies the phrase exists in the introspector script file.

Reply:
Added comment and validation in [6799942](https://github.com/oracle/weblogic-kubernetes-operator/pull/3408/commits/679994206d7619de43c35783ddc8f2e2914d7b82) and [8611f71](https://github.com/oracle/weblogic-kubernetes-operator/pull/3408/commits/8611f71526c5777d0fd300a7e637c552a5f52d96).

Reply:
Thanks